### PR TITLE
reduced lib size from 738,548 -> 438,991

### PR DESCRIPTION
**Pull request by jay-dee7**
Signed-off-by: Jasdeep Singh <jasdeepsingh.uppal@gmail.com>

### CODE REVIEW

**Comment by jay-dee7** [comment]
@Arachnid tiny improvements on #26

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "lint": "tslint -p tsconfig.json",
     "prepare": "npm run build",
     "prepublishOnly": "npm run test",
+    "sized": "browserify src/internal/index.js -p [tsify --noImplicitAny] | wc -c",
```
Comment (Arachnid) [inline package.json]: What's the purpose of this?

File: package.json [hunk 2]
```diff
@@ -31,23 +32,20 @@
...
-	"blakejs": "^1.1.0",
+    "blakejs": "^1.1.0",
     "cashaddrjs": "^0.3.8",
-    "eosjs-ecc": "^4.0.7",
-    "nem-sdk": "^1.6.7",
-    "ripple-address-codec": "^4.0.0",
-    "rskjs-util": "^1.0.3",
-    "stellar-base": "^2.1.2",
+    "crc": "^3.8.0",
+    "crypto-addr-codec": "^0.1.1",
```
Comment (Arachnid) [inline package.json]: Can you please pin this to a specific version?

Comment (jay-dee7) [reply package.json]: sure, can i include a yarn.lock file? or just by using the version would be okay?

File: package.json (hunk 1 above)
Comment (jay-dee7) [reply package.json]: i'll remove it. initially i was planning to use an `internal` directory instead of an npm package

**Comment by jay-dee7** [comment]
@Arachnid i'll check with all the packages and see if we can include their license without any issues

File: package.json (hunk 2 above)
Comment (Arachnid) [reply package.json]: Just using the version is fine.

**Comment by inspireme6** [comment]
Since @Arachnid is talking about transfer your repo, can you @jay-dee7 include the ss58, bs58, cashaddr and shajs files into the repo to keep it all in one place? Thanks

**Comment by jay-dee7** [comment]
@inspireme6 sure. can you make a PR to the crypto-addr-serialize repo. I see you've made some changes?

**Comment by inspireme6** [comment]
> @inspireme6 sure. can you make a PR to the crypto-addr-serialize repo. I see you've made some changes?

The only significant change was referencing the shajs modified in the package.json.. I had to add the latest files you uploaded (the stellar files) because they disappeared for some reason (?).  The modified shajs package is here https://github.com/inspireme6/sha256js

**Comment by jay-dee7** [comment]
> > @inspireme6 sure. can you make a PR to the crypto-addr-serialize repo. I see you've made some changes?
> 
> The only significant change was referencing the shajs modified in the package.json.. I had to add the latest files you uploaded (the stellar files) because they disappeared for some reason (?). The modified shajs package is here https://github.com/inspireme6/sha256js

thats my bad, i published it directly from my local work, i didn't push it to the repo. sorry

**Comment by inspireme6** [comment]
> > > @inspireme6 sure. can you make a PR to the crypto-addr-serialize repo. I see you've made some changes?
> > 
> > 
> > The only significant change was referencing the shajs modified in the package.json.. I had to add the latest files you uploaded (the stellar files) because they disappeared for some reason (?). The modified shajs package is here https://github.com/inspireme6/sha256js
> 
> @inspireme6 thats my bad, i published it directly from my local work, i didn't push it to the repo. sorry. can you please rebase it from master and update the PR?

I see you have pushed the changes already, it's necessary to rebase it anyways? I'm still new to git, so not sure.

**Comment by jay-dee7** [comment]
No its not. I'll do it dont worry 😁

**Comment by jay-dee7** [comment]
@Arachnid i've sent you an invitation for the github repo, please accept it and migrate the repository it to @ensdomains. Since i dont have permissions at ensdomains, i cant transfer.

**Comment by Arachnid** [comment]
@jay-dee7 I now have push access, but you need to add me as an owner before I can transfer it.

File: src/index.ts [hunk 3]
```diff
@@ -226,11 +230,8 @@ function b32encodeXemAddr(data: Buffer): string {
 }
 
 function b32decodeXemAddr(data: string): Buffer {
-  if(!isValid(data)) {
```
Comment (Arachnid) [inline src/index.ts]: This needs to remain in, unless you have something else to check validity.

**Comment by jay-dee7** [comment]
> @jay-dee7 I now have push access, but you need to add me as an owner before I can transfer it.

i couldn't make you an owner, i think that option is only for organizations, i've sent an invitation to your user account @Arachnid

File: src/index.ts (hunk 3 above)
Comment (jay-dee7) [reply src/index.ts]: adding it right back

**Comment by jay-dee7** [comment]
this validation check put me back in time :(
latest repo size as it shows in travis logs is 440,007

**Comment by Arachnid** [comment]
@jay-dee7 Thanks, I've accepted the transfer. Are you able to transfer the NPM package to me or to the ensdomains org?

**Comment by jay-dee7** [comment]
@Arachnid I'll let you know in next 20 minutes

**Comment by jay-dee7** [comment]
> @jay-dee7 Thanks, I've accepted the transfer. Are you able to transfer the NPM package to me or to the ensdomains org?

@Arachnid just added you as a maintainer for the npm package, please confirm it and i'll remove myself from the list

**Comment by jay-dee7** [comment]
@Arachnid i see that both, the repository and the npm package is under @ensdomains organization now. Please let me know if there's something, blocking this PR.

**Comment by inspireme6** [comment]
@Arachnid I just made a PR to the crypto-addr-serialize package that includes the rest of the encoders (bs58 and cashaddr) and includes the sha256 script which was part of the latest changes I made.

After @jay-dee7 PR gets merged, I'll make a PR to this repo that will update the files to include the changes from the updated crypto-addr-serialize package if it goes through.

**Pull request merged**